### PR TITLE
Minimal changes to make run_loop run with Xcode 10.2

### PR DESCRIPTION
**Pull request by JoeSSS**

### CODE REVIEW

**Comment by JoeSSS** [comment]
sure, we need to add tests, that's why it is WIP :)

btw, the CI won't work?

File: run_loop.gemspec [hunk 1]
```diff
@@ -56,7 +56,7 @@ tools like instruments and simctl.}
 
   s.add_development_dependency("rspec_junit_formatter", "~> 0.3")
   s.add_development_dependency("luffa", "~> 2.0")
-  s.add_development_dependency('bundler', '~> 1.6')
```
Comment (jmoody) [inline run_loop.gemspec]: Good.

File: lib/run_loop/xcode.rb [hunk 2]
```diff
@@ -26,6 +26,14 @@ def inspect
...
     end
 
+    # Returns a version instance for Xcode 10.2; used to check for the
+    # availability of features and paths to various items on the filesystem
+    #
+    # @return [RunLoop::Version] 10.2
+    def v102
+      fetch_version(:v102)
+    end
+
```
Comment (jmoody) [inline lib/run_loop/xcode.rb]: Expected to see corresponding `gte_102?` method:

```
    # Is the active Xcode version 10.2 or above?
    #
    # @return [Boolean] `true` if the current Xcode version is >= 10.2
    def version_gte_102?
      version >= v102
    end
```

Not blocking.

File: lib/run_loop/simctl.rb [hunk 3]
```diff
@@ -590,7 +590,15 @@ def device_key_is_watchos?(key)
 
     # @!visibility private
     def device_key_to_version(key)
-      str = key.split(" ").last
+      str = if key.include?(" ")
+              key.split(" ").last
+            else
+              # matches versions like 12-2 and 14-1-1
+              matches = /(?<version>\d+(-\d){1,2}\z)/.match(key)
```
Comment (jmoody) [inline lib/run_loop/simctl.rb]: Extract this to a constant in regex.rb.

File: lib/run_loop/simctl.rb [hunk 4]
```diff
@@ -590,7 +590,15 @@ def device_key_is_watchos?(key)
...
+              raise RuntimeError, "Cannot fetch the Simulator version from #{key}" if matches.nil?
```
Comment (jmoody) [inline lib/run_loop/simctl.rb]: I would prefer:

```
if matches.nil?
  raise ...
end


Not blocking.

**Comment by jmoody** [comment]
I will have a look at the Jenkins CI machine now.

**Comment by JoeSSS** [comment]
@jmoody I tried to install Xcode on my home laptop and it works on this branch without `xcrun: error: unable to find utility "simctl", not a developer tool or in PATH` error. Not sure how to reproduce it.. all the other comments seem to be addressed.

**Comment by jmoody** [comment]
Whoops.  We need to get this into the develop branch.  I will try to cherry pick this.

**Comment by jmoody** [comment]
Huh.  The UI is confusing.  `calabash:develop` => calabash/run_loop/develop.

Alright.

Good job.  I will try to push a release ASAP.

**Pull request merged**